sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptors/ScrubEtagInterceptor.java

```diff
@@ -0,0 +1,40 @@
+package com.azure.android.core.http.interceptors;
+
+import androidx.annotation.NonNull;
+
+import java.io.IOException;
+
+import okhttp3.Interceptor;
+import okhttp3.Response;
+
+/**
+ * Wraps any potential error responses from the service and applies post processing of the response's eTag header to
+ * standardize the value.
+ */
+public class ScrubEtagInterceptor implements Interceptor {
+    private static final String ETAG = "ETag";
+
+    /**
+     * The service is inconsistent in whether or not the ETag header value has quotes. This method will check if the
+     * response returns an ETag value, and if it does, remove any quotes that may be present to give the user a more
+     * predictable format to work with.
+     *
+     * @return an updated response with post processing steps applied.
+     */
+    @NonNull
+    @Override
+    public Response intercept(@NonNull Chain chain) throws IOException {
+        Response response = chain.proceed(chain.request());
+        String eTag = response.header(ETAG);
+
+        if (eTag == null) {
+            return response;
+        }
+
+        eTag = eTag.replace("\"", "");
+
+        return response.newBuilder()
```

Comment:
Two questions:

First, is this the only way to modify the headers? It seems inefficient but maybe it's not and that's just OkHTTP's design to always use a builder to make any modification to the request or response. I think this is the performance concern Adrian had mentioned.

Second, is the `header()` method case-insensitive? That is, if we retrieve the value from a header named "etag" and then replace it here using the name "ETag" will it work as expected or will it end up adding a second header?

Reply:
According to OkHttp's [documentation](https://square.github.io/okhttp/interceptors/#rewriting-requests), this is the way of modifying any headers on a request/response.

Reply:
As for the `header()` method, let me do some research on this.

Reply:
This method is case-insensitive.